conn.go

```diff
@@ -254,3 +296,26 @@ func (conn *Conn) writeLog(s string, a ...interface{}) {
 		conn.LogCh <- msg
 	}
 }
+
+func (conn *Conn) ping(ok chan bool) {
+	firstLoop := true
+	continueLoop := true
+	for continueLoop {
+		if conn.IsConnected() {
+			_, err := conn.write(ProtoReqPing, nil, 5)
+			if err != nil {
+				conn.writeLog("ping failed: %s", err)
+
+			} else {
+				conn.writeLog("ping! (%s:%d)", conn.host, conn.port)
+			}
+		} else {
+			continueLoop = false
+		}
+		if firstLoop {
+			firstLoop = false
+			ok <- true
+		}
+		time.Sleep(conn.PingInterval * time.Second)
```

Comment:
The Keep Alive should start with a sleep since otherwise the programmer had to time before enabling the keep-alive. (if the keep-alive is enabled immediately, a useless keep-alive package will be send)